src/storage/redis_metadata.cc

```diff
@@ -401,3 +401,73 @@ rocksdb::Status StreamMetadata::Decode(const std::string &bytes) {
 
   return rocksdb::Status::OK();
 }
+
+void SBChainMetadata::Encode(std::string *dst) {
+  Metadata::Encode(dst);
+
+  PutFixed16(dst, n_filters);
+  PutFixed16(dst, scaling);
+  PutFixed16(dst, expansion);
+}
+
+rocksdb::Status SBChainMetadata::Decode(const std::string &bytes) {
+  Slice input(bytes);
+  if (!GetFixed8(&input, &flags)) {
```

Comment:
@PragmaTwice Can it be like:

```c++
Status s = Metadata::Decode(bytes);
```

To avoid copy-paste original `Metadata::Decode`'s code?

Reply:
Sure.

Reply:
I have looked into this and it seems currently it is hard to reuse Metadata::Decode since the output Slice is missing. You can just copy code here and I will fix it later after it is merged.